Add basic snippets

rvanasa opened the pull request:
Includes basic Motoko snippets (adapted from #18).

krpeacock commented:
I'd recommend adding snippets for a for-loop iterating through items in an Array, as well as empty initializations for structures like Trie, HashMap, TrieMap, etc. 
```swift
let ${1: a0} = [];
for (${2: val} in ${1:a0}.vals()){
  ${3}
}

let ${1:t0} : Trie<${1:Text}, ${1:Nat}> = Trie.empty();
```

Some more advanced but useful cases would be snippets for inserting into structures, but this starts to get opinionated

```
let (new${1:Resource}, existing) = Trie.put(
  ${2:resource} ,  
  key(${3:keyName}), 
  ${4:equalityChecker},
  ${5:thingToInsert}
);
```

rvanasa commented:
Updated the `for-each` loop to call `.vals()`, and added a bunch of additional snippets for data structure initializations, conversions, etc. 

@kentosugama, any code snippet ideas for this initial PR?

kentosugama commented:
I'd probably give a simple example of an anonymous function. The list seems pretty complete though

krpeacock commented on snippets.json:
```diff
@@ -0,0 +1,356 @@
+    },
+    "Base Library Import": {
+        "prefix": [
+            "base-import"
```
```suggestion
            "import-base"
```

Since the prompt pops up when you start typing, it's better to start with the keyword that someone will organically start typing

krpeacock commented on snippets.json:
```diff
@@ -0,0 +1,356 @@
+    },
+    "Mutable Array Literal": {
+        "prefix": [
+            "var-array"
```
```suggestion
            "array-mut"
```

krpeacock commented on snippets.json:
```diff
@@ -0,0 +1,356 @@
+    },
+    "New Array": {
+        "prefix": [
+            "new-array"
```
```suggestion
            "array-new"
```

krpeacock commented on snippets.json:
```diff
@@ -0,0 +1,356 @@
+    },
+    "New Hash Map": {
+        "prefix": [
+            "new-hash-map"
```
```suggestion
            "hash-map-new"
```

krpeacock commented on snippets.json:
```diff
@@ -0,0 +1,356 @@
+    },
+    "New Trie": {
+        "prefix": [
+            "new-trie"
```
```suggestion
            "trie-new"
```

krpeacock commented on snippets.json:
```diff
@@ -0,0 +1,356 @@
+    },
+    "New Trie Map": {
+        "prefix": [
+            "new-trie-map"
```
```suggestion
            "trie-map-new"
```

krpeacock commented:
I've created a few suggestions, but as a basic principal, I'd say to use suffixing for modifiers like `"new-trie-map"`, so that the keyword, `trie` is what people need to start typing. There are a few more cases like this

rvanasa commented:
Updated `import-base` and `array-mut`. I intentionally used `new-*` prefixes so that people could type `new` and view alternate data structures which could work for a given situation. However, I'll defer to your judgement if you think it would be better to use suffixes in this case.

rvanasa commented:
Final call for snippet suggestions; merging tomorrow.
